src/lib/cache/subCache.cpp

```diff
@@ -1198,33 +1204,48 @@ void subCacheSync(void)
   while (cSubP != NULL)
   {
     CachedSubSaved* cssP = savedSubV[cSubP->subscriptionId];
-    if ((cssP != NULL) && (cssP->lastNotificationTime <= cSubP->lastNotificationTime))
+
+    if (cssP != NULL)
     {
-      // cssP->lastNotificationTime older than what's currently in DB => throw away
-      cssP->lastNotificationTime = 0;
+      if (cssP->lastNotificationTime <= cSubP->lastNotificationTime)
+      {
+        // cssP->lastNotificationTime is older than what's currently in DB => throw away
+        cssP->lastNotificationTime = 0;
+      }
+
+      if (cssP->lastFailure < cSubP->lastFailure)
```

Comment:
I understand this is going to change, so maybe it is better to review on the final status, in a next PR.